dvc/exceptions.py

```diff
@@ -7,6 +7,10 @@ class DvcException(Exception):
     """Base class for all dvc exceptions."""
 
 
+class InvalidArgumentError(ValueError, DvcException):
+    """Thrown if arguments are invalid."""
```

Comment:
Nice one! Should we use it in otherplaces like `dvc/repo/__init__.py` and `dvc/repo/reproduce.py` too?

Reply:
I'd opt for it on a different PR. :slightly_smiling_face:

Reply:
@skshetry Sure, let's not forget about it though :slightly_smiling_face: